File: app/MoviesApp/app/src/main/java/com/example/moviesapp/data/movies/MoviesApi.java
package com.example.moviesapp.data.movies;

import com.example.moviesapp.data.movies.dto.MoviesDto;

import io.reactivex.Single;
import retrofit2.http.GET;

public interface MoviesApi {

    @GET("/api/v1/movies")
    Single<MoviesDto> movies();
}

File: app/kotlin/MoviesApp/app/src/main/java/com/example/moviesapp/data/movies/MoviesApi.kt
package com.example.moviesapp.data.movies

import com.example.moviesapp.data.movies.dto.MoviesListDto
import io.reactivex.Single
import retrofit2.http.GET

internal interface MoviesApi {

    @GET("/api/v1/movies")
    fun movies(): Single<MoviesListDto>
}